File: python-server/src/redis/redisConnection.py
# src/redis/redis_connection.py
import redis
import os
from dotenv import load_dotenv

load_dotenv()  # load .env variables

# Create a single Redis connection client
redis_client = redis.Redis(
    host=os.getenv("REDIS_HOST"),
    port=int(os.getenv("REDIS_PORT", 6379)),
    password=os.getenv("REDIS_PASSWORD"),
    decode_responses=True  # returns str instead of bytes
)

File: node-server/src/redis/redisConnection.ts
import { createClient, RedisClientType } from 'redis';

// Create and export the Redis client
export const redisClient: RedisClientType = createClient({
    username: 'default', // default for Redis Cloud
    password: process.env.REDIS_PASSWORD, // store in .env
    socket: {
        host: process.env.REDIS_HOST, // store in .env
        port: Number(process.env.REDIS_PORT) || 6379
    }
});

console.log('REDIS_HOST:', process.env.REDIS_HOST);
console.log('REDIS_PORT:', process.env.REDIS_PORT);

// Listen for errors
redisClient.on('error', (err) => console.error('Redis Client Error', err));

// Connect to Redis
export const connectRedis = async (): Promise<void> => {
    try {
        await redisClient.connect();
        console.log('✅ Connected to Redis Cloud');
    } catch (err) {
        console.error('❌ Redis connection failed:', err);
    }
};